editors/sc-ide/core/main.cpp

```diff
@@ -98,7 +98,16 @@ int main( int argc, char *argv[] )
     // Go...
     MainWindow *win = new MainWindow(main);
 
-    app.setWindowIcon(QIcon("qrc:///icons/sc-ide-svg"));
+    // Unfortunately, the SVG icon shows up as a tiny dot on some Linux window
+    // managers (see #3905, #2646). Best we can do here is a PNG.
+    // app.setWindowIcon(QIcon(":icons/sc-ide-svg"));
+    QIcon icon;
+    icon.addFile(":icons/sc-ide-512");
+    icon.addFile(":icons/sc-ide-128");
+    icon.addFile(":icons/sc-ide-64");
+    icon.addFile(":icons/sc-ide-32");
+    icon.addFile(":icons/sc-ide-16");
+    app.setWindowIcon(icon);
```

Comment:
i'm confused -- we have code for this both here and in main_window.cpp: 

https://github.com/supercollider/supercollider/blob/develop/editors/sc-ide/widgets/main_window.cpp#L229-L240

why is the code needed in both places, and why doesn't it match?